src/Storages/MergeTree/checkDataPart.cpp

```diff
@@ -166,6 +166,9 @@ IMergeTreeDataPart::Checksums checkDataPart(
     bool require_checksums,
     std::function<bool()> is_cancelled)
 {
+    if (!data_part->isStoredOnDisk())
```

Comment:
Why not check `data.bin`. Maybe bit flip happened.

Reply:
I've added a check, but I have no idea how write negative test for it. We have to simulate data corruption in ram or during transfering over network in integration test.